lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -975,6 +1001,68 @@ def configure_network(self, vm_obj):
                 self.configspec.deviceChange.append(nic)
                 self.change_detected = True
 
+    def configure_vapp_properties(self, vm_obj):
+        if len(self.params['vapp_properties']) == 0:
+            return
+
+        for x in self.params['vapp_properties']:
+            if not x.get('id'):
+                self.module.fail_json(msg="id is required to set vApp property")
+
+        spec = vim.vApp.VmConfigSpec()
+
+        # This is primarily for vcsim/integration tests, unset vAppConfig was not seen on my deployments
+        orig_spec = vm_obj.config.vAppConfig if vm_obj.config.vAppConfig else spec
+
+        vappProperties = dict((x.id, x) for x in orig_spec.property)
+        propsToChange = dict((x['id'], x) for x in self.params['vapp_properties'])
+
+        # each property must have a unique key
+        # init key counter with max value + 1
+        allKeys = [x.key for x in orig_spec.property]
+        newPropIndex = max(allKeys) + 1 if allKeys else 0
+
+        for k, v in propsToChange.items():
+            tmp = vim.VAppPropertySpec()
+            if k in vappProperties.keys():
+                tmp.operation = v.get('operation', 'edit')
+                if tmp.operation == 'remove':
+                    tmp.removeKey = vappProperties[k].key
+                else:
+                    # this is 'edit' branch
+                    # updating only required properties
+                    tmp.info = vappProperties[k]
+                    try:
+                        for infoPropName, infoPropVal in v.items():
+                            setattr(tmp.info, infoPropName, infoPropVal)
```

Comment:
Can you modify this method so the property get updated only if needed (if changed have to be applied) ?

Reply:
agreed

Reply:
Fixed